azavea/climate-change-lab
# Feature/new & edit project workflow

## Overview

Create or edit a project and save all, too! It can accept projects with no input.

The edit project button only shows up when you're in a project.

### Demo

<img width="422" alt="screen shot 2016-09-27 at 2 19 47 pm" src="https://cloud.githubusercontent.com/assets/10568752/18886354/8f0e1fbe-84bd-11e6-8275-3685f1e0359e.png">
<img width="545" alt="screen shot 2016-09-27 at 2 20 08 pm" src="https://cloud.githubusercontent.com/assets/10568752/18886358/90992ba8-84bd-11e6-97f6-e055a9dfb997.png">
<img width="509" alt="screen shot 2016-09-27 at 2 21 10 pm" src="https://cloud.githubusercontent.com/assets/10568752/18886395/acbf8b92-84bd-11e6-821a-3cfe8c16528b.png">
<img width="609" alt="screen shot 2016-09-28 at 10 13 50 am" src="https://cloud.githubusercontent.com/assets/10568752/18917092/af0456ec-8564-11e6-960e-50aba4476f4a.png">



### Notes

Rather than modals I made add/edit project into fullpage views. Flexible.
Light styling.
When selecting models, if you click "update" it will submit the add/edit form which is not desired. Opened issue #85.

## Testing Instructions

You can only access the lab with a project id in the URL, now.

Create a project from the dashboard and it should lead you directly to it.
Else, localhost:3100/lab/your_existing_project_id_here

Edit: from lab/project use the navbar user dropdown, project settings. OR localhost:3100/edit/project_id_here

You can find your projects by using Augury debugging and looking into your projectService which stores an array of projects.

Closes #80, #79, #86
Opens issues #85
